File: 3834-minimum-operations-to-convert-all-elements-to-zero/minimum-operations-to-convert-all-elements-to-zero.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        stack = []
        ans = 0

        for num in nums:
            while stack and stack[-1] > num:
                stack.pop()

            if (num != 0) and (not stack or stack[-1] < num):
                ans += 1
                stack.append(num)

        return ans

File: 3834-minimum-operations-to-convert-all-elements-to-zero/minimum-operations-to-convert-all-elements-to-zero.cs
public class Solution {
    public int MinOperations(int[] nums) {
        var stack = new List<int>();
        int ans = 0;

        foreach (int num in nums) {
            while (stack.Count > 0 && stack[stack.Count - 1] > num) {
                stack.RemoveAt(stack.Count - 1);
            }
            if ((num != 0) && (stack.Count == 0 || stack[stack.Count - 1] < num)) {
                ans++;
                stack.Add(num);
            }
        }
        
        return ans;
    }
}